preview-maker.js

```diff
@@ -205,45 +201,34 @@ const makePrevDesktop = async (themeBuffer) => {
   const variables = theme.variables();
 
   let colors = [];
-  const dialogsBg = rgbToHsl(theme.resolveVariable(`dialogsBg`));
+  let grey = 0;
   for (const variable of variables) {
     const elements = getElementsByClassName(preview, variable);
     const color = theme.resolveVariable(variable);
     for (const element of elements) {
       fill(element, color);
     }
-    const chooseHsl = rgbToHsl(color);
-    let hueDifference = Math.abs(chooseHsl.hue - rgbToHsl(dialogsBg).hue);
-    if (hueDifference > 180) {
-      hueDifference = 360 - hueDifference;
-    }
-    let saturationDifference = Math.abs(
-      chooseHsl.saturation - dialogsBg.saturation
-    );
-    if (
-      hueDifference > 2 &&
-      chooseHsl.saturation > 0.04 &&
-      saturationDifference > 0.02
-    ) {
-      colors.push(Math.round(chooseHsl.hue));
+    const dialogsBg = rgbToHsl(theme.resolveVariable(`dialogsBg`));
+    const colorHsl = rgbToHsl(color);
+    if (!eqv(dialogsBg, colorHsl) && elements.length > 0) {
+      if (isNotGrey(colorHsl)) {
+        colors.push(Math.round(colorHsl.hue));
+      } else {
+        grey += 1;
+      }
```

Comment:
You're currently doing the same thing for Android and Desktop themes: calculating the amount of gray variables. This better be done in `calculateAccentColor`, which is supposed to calculate the accent color regardless of the underlying theme format. In other words, this function only needs to select variables which are used in the preview and aren't the theme's background, everything else needs to be dome in `calculateAccentColor`.